src/cable/payload.cr

```diff
@@ -1,82 +1,118 @@
 module Cable
-  class Payload
+  struct Payload
+    include JSON::Serializable
+    include JSON::Serializable::Unmapped
     alias RESULT = String | Int64 | Hash(String, RESULT)
     alias PARAMS = Hash(String, RESULT)
 
-    getter json : String
-    getter action : String
-    getter command : String?
-    getter identifier : String
-    getter channel : String?
-    getter channel_params : Hash(String, Cable::Payload::RESULT) = Hash(String, RESULT).new
-    getter data : Hash(String, Cable::Payload::RESULT) = Hash(String, RESULT).new
-
-    def initialize(@json : String)
-      @parsed_json = JSON.parse(@json)
-      @action = ""
-      @is_action = false
-      @command = @parsed_json["command"].as_s
-      @identifier = @parsed_json["identifier"].as_s
-      @channel = process_channel
-      @channel_params = process_channel_params
-      @data = process_data
+    module IdentifierConverter
+      def self.from_json(value : JSON::PullParser) : Indentifier
+        key = value.read_string
+        i = Indentifier.from_json(key)
+        i.key = key
+        i
+      end
+
+      def self.to_json(value : Indentifier, json : JSON::Builder) : Nil
+        json.string(value.to_json)
```

Comment:
```suggestion
        json.string(value.key)
```

I just realized because we store the original formatting, we should just convert back to this.. This is because you may have sent in an identifier like `{"channel": "RoomChannel"}`, but Crystal's `to_json` would make `{"channel":"RoomChannel"}` (space removed), and this would end up being a different key stored in the connections hash.